src/vga/src/cursor.rs

```diff
@@ -0,0 +1,58 @@
+//! Dealing with the VGA cursor.
+//!
+//! The VGA cursor is handled through a system of 'indexed registers.' This means that you take a
+//! location, `CRT_PORT` in this case, and write the index you'd like to access to it. You can then
+//! access `CRT_PORT + 1` to read/write to that index. Within these functions, we've called these
+//! two ports `crt_index` and `crt_io`, respectively. We create these ports within each function,
+//! rather than trying to safe some kind of static value, because statics aren't safe, and it's not
+//! like the `Port`s are expensive to create in the first place.
+//!
+//! This module has two functions, one of which initializes the cursor in the first place, and the
+//! second which updates it to a particular location. `initialize()` must be called before you call
+//! `set()` or `set()` can't do its job.
+//!
+//! References:
+//!
+//! * [OSDev wiki](http://wiki.osdev.org/Text_Mode_Cursor)
+//! * [Wikipedia](http://wiki.osdev.org/VGA_Hardware)
+
+use cpuio::Port;
+
+const CRT_PORT: u16 = 0x3D4;
+
+/// Initializes the cursor.
+///
+/// This function must be called in order to use the cursor, otherwise, it will not work.
+pub fn initialize() {
+    let mut crt_index: Port<u8> = unsafe { Port::new(CRT_PORT) };
+    let mut crt_io: Port<u8> = unsafe { Port::new(CRT_PORT + 1) };
+
+    // Set up cursor start register (0x0Ah)
+    // Bits 0-4: Scanline start (where the cursor beings on the y axis)
+    // Bit    5: Visibility status (0 = visible, 1 = invisible)
+    crt_index.write(0b1010);
```

Comment:
So I believe the register itself is a full byte, at least according this this [http://www.stanford.edu/class/cs140/projects/pintos/specs/freevga/vga/crtcreg.htm#0A](doc). That doc was the best I found while trying to figure out the cursor magic, and maybe the best for general VGA register info. The OSDev doc is pretty lacking, only describing the set location bit, one of the reference docs has more info but I found it hard to read. I recommend at least linking the Stanford doc along with the OSDev doc for clarity.